packages/components/datepicker/src/components/osds-datepicker/core/controller.ts

```diff
@@ -0,0 +1,55 @@
+import type { OsdsDatepicker } from '../osds-datepicker';
+import { OdsLogger } from '@ovhcloud/ods-common-core';
+
+/**
+ * common controller logic for input component used by the different implementations.
+ * it contains all the glue between framework implementation and the third party service.
+ */
+class OdsDatepickerController {
+  private readonly component: OsdsDatepicker;
+  private readonly logger = new OdsLogger('OdsDatepickerController');
+
+  constructor(component: OsdsDatepicker) {
+    this.component = component;
+  }
+
+  beforeInit(): void {
+    this.logger.debug('[datepicker]', 'beforeInit', this.component.value);
+    this.onChange(this.component.value);
+  }
+
+  onFocus() {
+    this.component.hasFocus = true;
+    this.component.emitFocus();
+  }
+
+  onChange(newValue: Date | undefined | null, value?: Date | null) {
+    this.logger.debug(`[datepicker=${newValue}]`, 'value changed', { newValue, value });
+    if(!this.component.disabled) {
+      if (newValue === undefined || newValue === null || isNaN(newValue.getTime())) {
```

Comment:
seems like a very simplist way to check if the value is a valid Date, what about external lib validation method ?

Reply:
This is only a double-check of the value /// main Date validation is done on user input through Datepicker’s parseDate function with format & length verification (check Listener of odsValueChange)